src/dreyfus_index.erl

```diff
@@ -157,9 +159,43 @@ handle_call(info, _From, State) -> % obsolete
     Reply = info_int(State#state.index_pid),
     {reply, Reply, State}.
 
+handle_cast({ddoc_updated, DDocResult}, State) ->
+    #state{dbname = DbName, index=#index{sig=CurrentSig}, waiting_list = WaitList} = State,
+    IndexDeleted = case DDocResult of
+        {not_found, deleted} ->
+            true;
+        {ok, DDoc} ->
+            Index = get_index_with_sig(DDoc, CurrentSig),
+            case Index of
+                [] ->
+                    true;
+                _ ->
+                    false
+            end
+        end,
+        case IndexDeleted of
+            true ->
+                case WaitList of
+                    [] ->
+                        self() ! {index_deleted},
+                        {noreply, State};
+                    _ ->
+                        erlang:send_after(?INDEX_SHUTDOWN_DELAY, self(), {index_deleted}),
```

Comment:
Basically if the waitlist is empty, it could either be waiting for the indexing to complete or it could be waiting for the end user request. Other option is to check and find out if the waitlist is the result of the end user request or the Ken request, but I don't think it's worth the effort.. just wait for one minute to let the index to finish the end user requests if any in the wait list.